packages/eslint-plugin/lib/config/typescript.ts

```diff
@@ -1,76 +1,83 @@
+import js from "@eslint/js";
 import type { Linter } from "eslint";
-import { typescriptFiles } from "../utils/patterns";
+import { concat } from "eslint-flat-config-utils";
+import globals from "globals";
+import tseslint from "typescript-eslint";
+import { typeScriptFiles } from "../utils/patterns.ts";
 
-const config: Linter.Config = {
-    overrides: [
-        {
-            files: typescriptFiles,
-            parser: "@typescript-eslint/parser",
-            plugins: ["@typescript-eslint"],
-            extends: [
-                "plugin:@typescript-eslint/eslint-recommended",
-                "plugin:@typescript-eslint/recommended"
-            ],
-            rules: {
-                // @typescript-eslint/recommended disables
-                "@typescript-eslint/no-non-null-assertion": "off",
+const config: Linter.FlatConfig[] = await concat(
+    js.configs.recommended,
```

Comment:
I am not sure this is equivalent to:

```
extends: [
    "plugin:@typescript-eslint/eslint-recommended",
    "plugin:@typescript-eslint/recommended"
]
```

Reply:
This is how they recommend doing it now in the typescript-eslint docs. Instead of re-providing the ESLint rules, they just have you import them from ESLint directly.